spring-amqp/src/main/java/org/springframework/amqp/core/Queue.java

```diff
@@ -145,6 +151,16 @@ public String getDeclaredName() {
 		return this.declaredName;
 	}
 
+	/**
+	 * Return the name provided to the constructor or the broker-generated name
+	 * if that name is an empty String.
+	 * @return the name.
+	 * @since 2.0.4
+	 */
+	public String getActualName() {
+		return StringUtils.hasText(this.name) ? this.name : this.declaredName;
```

Comment:
Can't we have a property on the matter and update it when `setDeclaredName()` is called?
That's might be over-optimization, but still - curious what is your opinion :smile: